erpnext/selling/page/point_of_sale/pos_controller.js

```diff
@@ -559,8 +559,10 @@ erpnext.PointOfSale.Controller = class {
 
 				item_row = this.frm.add_child('items', new_item);
 
-				if (field === 'qty' && value !== 0 && !this.allow_negative_stock)
-					await this.check_stock_availability(item_row, value, this.frm.doc.set_warehouse);
+				if (field === 'qty' && value !== 0 && !this.allow_negative_stock) {
+					const qty_needed = field === 'qty' ? value * item_row.conversion_factor : item_row.qty * value;
```

Comment:
It will enter inside if block only if the field is `qty`. Whats the point of adding a conditional operator here?

Reply:
If multi uom case it fetch conversion factor otherwise value always be 1 for example explain above if that remove it restrict user to use multiple uom setup. Default uom qty get validated so example 1 kg and product is 50g so imagine 800g = .8kg item available so it should allow to sell 160 unit of 50g. But it say item not available thats reason it added here

Reply:
But my point is that condition is already validated on if condition, why it is needed inside the lf block as well?